lib/api_repo.js

```diff
@@ -713,24 +734,29 @@ ApiRepo.prototype._makeProtocFunc = function _makeProtocFunc(opts, language) {
     fs.mkdirsSync(outDir);
     var args = [];
     if (language === 'go') {
-      args.push(
-          '--' + language + '_out=plugins=grpc:' + outDir,
-          protoPath);
+      args.push('--' + language + '_out=plugins=grpc:' + outDir);
     } else {
-      var pluginBin = that.depBins['grpc_' + language + '_plugin'];
-      if (_.has(that.overridePlugins, language)) {
-        pluginBin = that.depBins[that.overridePlugins[language]];
+      if (language === 'php') {
+        // The php protoc plugin will generate php files for imported protos
+        // (which will be common APIs) without this flag. Other plugins have
+        // this behaviour by default
+        args.push('--' + language + '_out=skip-imported=true:' + outDir, '-I.');
+      } else {
+        args.push('--' + language + '_out=' + outDir, '-I.');
       }
-      args.push('--' + language + '_out=' + outDir, '-I.');
       _.each(that.includePath, function(aPath) {
         args.push('-I' + aPath);
       });
       if (!opts.buildCommonProtos) {
-        args.push('--grpc_out=' + outDir,
-                  '--plugin=protoc-gen-grpc=' + pluginBin);
+        if (language != 'php') {
+          // The php protoc plugin will fail when passed this argument
+          args.push('--grpc_out=' + outDir);
+        }
+        var pluginBin = that.depBins[that._getPluginName(language, that.overridePlugins)];
+        args.push('--plugin=protoc-gen-grpc=' + pluginBin);
```

Comment:
PHP doesn't use protoc-gen-grpc - I will restructure so that this is not included in the PHP case